### ISSUE TITLE
[java] CollectionTypeMismatch: False negative when collection is declared using 'var' local variable type inference

### ISSUE THREAD

**Issue by leemeii** [pmd/pmd#6526]
**Affects PMD Version:** 7.22.0

**Rule:**[CollectionTypeMismatch](https://pmd.github.io/pmd/pmd_rules_java_errorprone.html#collectiontypemismatch)

**Description:**
The **CollectionTypeMismatch** rule fails to trigger when a generic collection is declared using Java's local variable type inference (```var```).
Even though the right-hand side clearly initializes a ```HashSet<Integer>```, PMD does not perform proper generic type inference on var, causing it to miss obvious type-mismatch bugs in ```contains()```, ```remove()```, etc. 

**Code Sample demonstrating the issue:**

```java
import java.util.HashSet;

public class a {

    HashSet<Integer> normalSet = new HashSet<Integer>();

    public void testNormal(String f) {
        if (normalSet.contains(f)) { //<-TP
            System.out.println("Found");
        }
    }

    public void testFalseNegative(String f) {
        var inferredSet = new HashSet<Integer>();

        if (inferredSet.contains(f)) { //<-should report(FN)
            System.out.println("Logically impossible, but PMD is silent.");
        }
    }
}

```

**Expected outcome:**
PMD should report ```CollectionTypeMismatch``` for both cases .
But currently it only reports the explicit type case (```normalSet```), completely missing the var case (```inferredSet```).
This is a clear **false negative** — the rule works correctly only when the type is declared explicitly.

**Running PMD through:** *CLI*

**Comment by UncleOwen** [comment]
Both PMD 7.22.0 and 7.23.0 report both violations:
```
./a.java:8:     CollectionTypeMismatch: Object of type 'java.lang.String' cannot be in collection expecting 'java.lang.Integer'
./a.java:16:    CollectionTypeMismatch: Object of type 'java.lang.String' cannot be in collection expecting 'java.lang.Integer'
```
